### ISSUE TITLE
Unannotated decorator causes "error: Expected N positional arguments"

### ISSUE THREAD

**Issue by Ceda-EI** [microsoft/pyright#2233]
**Describe the bug**
In the following code example, Pyright expects only one positional argument while it should expect two.

```python
def decorator(func):
    def wrapper(a, b):
        print(func(a) + b)
    return wrapper


@decorator
def foo(a):
    return a


foo(1, 2)
```

**To Reproduce**
- Copy the above code into a file.
- Run pyright on that file.

**Expected behavior**
No Errors

**Screenshots or Code**

```
$ cat pyright-issue.py
def decorator(func):
    def wrapper(a, b):
        print(func(a) + b)
    return wrapper


@decorator
def foo(a):
    return a


foo(1, 2)
$ python pyright-issue.py
3
$ pyright pyright-issue.py
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/pyright-issue.py
  /tmp/pyright-issue.py:12:8 - error: Expected 1 positional argument (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos
Completed in 0.629sec
$
```
**VS Code extension or command-line**
I am running pyright from the command line.

```
$ pyright --version
pyright 1.1.163
```
**Additional context**
The issue is not reproducible if type annotations are added. The following version does **not** show an error.

```python
def decorator(func):
    def wrapper(a: int , b: int):
        print(func(a) + b)
    return wrapper


@decorator
def foo(a):
    return a


foo(1, 2)
```
